src/module.c

```diff
@@ -5711,86 +5713,60 @@ robj **moduleCreateArgvFromUserFormat(const char *cmdname, const char *fmt, int
- *     * `S` -- Run the command in a script mode, this means that it will raise
- *              an error if a command which are not allowed inside a script
- *              (flagged with the `deny-script` flag) is invoked (like SHUTDOWN).
- *              In addition, on script mode, write commands are not allowed if there are
- *              not enough good replicas (as configured with `min-replicas-to-write`)
- *              or when the server is unable to persist to the disk.
- *     * `W` -- Do not allow to run any write command (flagged with the `write` flag).
- *     * `M` -- Do not allow `deny-oom` flagged commands when over the memory limit.
- *     * `E` -- Return error as RedisModuleCallReply. If there is an error before
- *              invoking the command, the error is returned using errno mechanism.
- *              This flag allows to get the error also as an error CallReply with
- *              relevant error message.
- * * **...**: The actual arguments to the Redis command.
- *
- * On success a RedisModuleCallReply object is returned, otherwise
- * NULL is returned and errno is set to the following values:
- *
- * * EBADF: wrong format specifier.
- * * EINVAL: wrong command arity.
- * * ENOENT: command does not exist.
- * * EPERM: operation in Cluster instance with key in non local slot.
- * * EROFS: operation in Cluster instance when a write command is sent
- *          in a readonly state.
- * * ENETDOWN: operation in Cluster instance when cluster is down.
- * * ENOTSUP: No ACL user for the specified module context
- * * EACCES: Command cannot be executed, according to ACL rules
- * * ENOSPC: Write or deny-oom command is not allowed
- * * ESPIPE: Command not allowed on script mode
- *
- * Example code fragment:
- * 
- *      reply = RedisModule_Call(ctx,"INCRBY","sc",argv[1],"10");
- *      if (RedisModule_CallReplyType(reply) == REDISMODULE_REPLY_INTEGER) {
- *        long long myval = RedisModule_CallReplyInteger(reply);
- *        // Do something with myval.
- *      }
- *
- * This API is documented here: https://redis.io/topics/modules-intro
- */
-RedisModuleCallReply *RM_Call(RedisModuleCtx *ctx, const char *cmdname, const char *fmt, ...) {
+static int validateACLS(RedisModuleCtx *ctx, client *c, int flags, RedisModuleCallReply **reply)
+{
+    int acl_errpos;
+    int acl_retval = ACLCheckAllPerm(c,&acl_errpos);
+    if (acl_retval != ACL_OK) {
+        addACLLogEntry(c,acl_retval,(c->flags & CLIENT_MULTI) ? ACL_LOG_CTX_MULTI : ACL_LOG_CTX_TOPLEVEL,acl_errpos,NULL,NULL);
+        if (flags & REDISMODULE_ARGV_CALL_REPLIES_AS_ERRORS) { /* return errors as RedisModuleCallReply object */
+            switch (acl_retval) {
+                case ACL_DENIED_CMD:
+                {
+                    sds msg = sdscatfmt(sdsempty(),
+                                        "-NOPERM this user has no permissions to run "
+                                        "the '%s' command", c->cmd->fullname);
```

Comment:
If error starts with `-`, `callReplyCreateError` will not add `\r\n` at tail.
```suggestion
                                        "the '%s' command\r\n", c->cmd->fullname);
```

Reply:
so this code was reverted for now.